doc/userdoc/contribute/styleguide/styleguide.rst

```diff
@@ -0,0 +1,649 @@
+
+In general, we try to follow the pattern:
+
+* First heading: ===
+* Second heading: ---
+* Third heading: ~~~
+* Fourth heading: ^^^
+
+"Double quotes"
+~~~~~~~~~~~~~~~
+
+We use double quotes for strings in code, for example,  ``nest.Create("iaf_psc_alpha")``. This applies to
+reStructuredText files as well as Python and C++ code. This rule is based on PEP 257, which (only) dictates the use of
+double quotes in triple quoted strings; for consistency, double quotes are used throughout the codebase.
+
+Double or single quotes should not be used to emphasize important concepts in the text.
+
+.. _sec_dbltick:
+
+\``Double backticks\``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Use double backticks for all code and command related terms, such as function call examples, paths, variables, and parameters.
+In addition, meta and special characters (such as the ampersand ``&``) should also be written in double backticks.
+
+For example::
+
+    ``nest.Create("iaf_psc_alpha")``
+
+    ``/path/to/source/file.rst``
+
+    "The key ``rule`` in the connectivity specification dictionary ``conn_spec`` . . . "
+
+
+\**Strong emphasis\** vs \*emphasis\*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Avoid using strong emphasis (boldface) in texts. If you want text to stand out use an appropriate :ref:`admonition <sec_admonition>`.
+
+Use the plain emphasis (italics) to signify context differences *within* a text.
+
+
+NumPy style docstrings
+----------------------
+
+In PyNEST code, we follow most of the rules for NumPy style docstrings as
+`explained in the NumPy style guide <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+
+However, we use different formatting marks than what is stated in their guide. See section on
+:ref:`double back ticks <sec_dbltick>`.
+
+If you are contributing to the :doc:`PyNEST API <example_template>`, make sure you carefully read the NumPy guide, along
+with this one.
+
+
+Bibliography style
+------------------
+
+The reStructuredText bibliography style is used throughout the documentation so that links
+are autogenerated and a consistent format is used.
+
+For in-text citations, we use the reStructuredText numeric style ``[1]_``.
+
+For example:
+
+    The following example is based on Smith [1]_.
+    Sanders et al. [2]_ contains a technically detailed example.
+
+Please ensure your reference follows the following guidelines.
+
+*  Do not add formatting markup such as italics, bold, or underline.
+*  Use a period after every section of bibliography.
+*  Use et al. for references with more than five authors.
+*  Put surname before first name for all authors.
+*  Do not put commas after surname.
+*  Use inital for first name of all authors.
+*  Put year, in parentheses, after authors.
+*  Write article titles in sentence case.
+*  Write the full title of journal.
+*  Insert a colon between volume and page-range.
+*  Add issue in parentheses after volume (optional).
+*  Include a linked DOI, if available.
+
+.. code-block:: none
```

Comment:
Is this line supposed to be here? References below are not rendered this way.

Reply:
Yes, I use the none code-block for the example references so I don't get the highlighting that `::` produces. It's not meant to show you how to create references - it's just in the raw reStructuredText file to remove syntax highlighting. See below for what it looks like with the double colon:

![image](https://user-images.githubusercontent.com/3313313/117856270-b1e97f00-b28b-11eb-8af7-0000f4d67948.png)